### PR TITLE
OWLS 79243  :  Helm 3/2 support for Operator Integration Tests

### PR DESCRIPTION
**Pull request by anpanigr**
Added conditional Logic to support both Helm 3.x and Helm 2.x type of commands while running Operator Integration Tests.  Most of changes are around 
   (a) helm install  (b) helm delete/uninstall   (c) helm list  (d) helm upgrade 
 
All the command in Helm 3 need a Namespace explicit argument, (--name) has been removed in install command . 

Ref :  https://itnext.io/breaking-changes-in-helm-3-and-how-to-fix-them-39fea23e06ff

Removed Helm Environment variable dependency from the Test. Now the BaseTest will determine the Helm Version)  automatically

External Jenkin Job (passed) with an intermittent failure

https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-quicktest/1049/
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-quicktest/1050/

Based on composite results, the tests failures are not related to helm. So the PR is ready for review

New external Jenkin Run:  https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-quicktest/1059/        ( one intermittent failure )

### CODE REVIEW

File: integration-tests/src/test/java/oracle/kubernetes/operator/utils/LoadBalancer.java [hunk 1]
```diff
@@ -222,23 +242,11 @@ private void createTraefikIngress() throws Exception {
...
-
-    executeHelmCommand(cmd2);
-
-    String cmd3 =
-        "helm install appscode/voyager --name voyager-operator --version 7.4.0 --namespace voyager "
-            + "--set cloudProvider=baremetal --set apiserver.enableValidatingWebhook=false";
-    LoggerHelper.getLocal().log(Level.INFO, "Executing Install voyager operator cmd " + cmd3);
-
-    executeHelmCommand(cmd3);
+    String cmdLb = "Empty String";
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/kubernetes/operator/utils/LoadBalancer.java]: fix initialization

File: integration-tests/src/test/java/oracle/kubernetes/operator/utils/LoadBalancer.java [hunk 2]
```diff
@@ -294,23 +306,31 @@ private void upgradeVoyagerNamespace() throws Exception {
...
-        .append(lbMap.get("domainUID"))
-        .append(" --set ")
-        .append("wlsDomain.clusterName=")
-        .append(lbMap.get("clusterName"))
-        .append(" --set ")
-        .append("voyager.webPort=")
+    cmd.append(chartDir).append(" && ");
+
+    if (BaseTest.HELM_VERSION.equals("V2")) {
+      cmd.append(" helm install ingress-per-domain ")
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/kubernetes/operator/utils/LoadBalancer.java]: same comment as above

File: integration-tests/src/test/java/oracle/kubernetes/operator/BaseTest.java [hunk 3]
```diff
@@ -82,6 +84,17 @@
...
+    if (HELM_VERSION == null) {
+      HELM_VERSION = "V2";
+    }
+ 
+    VOYAGER_VERSION = System.getenv("VOYAGER_VERSION");
+    if (VOYAGER_VERSION == null) {
+      VOYAGER_VERSION = "10.0.0";
+    }
+
+    System.out.println("(PANI) HELM VERSION IS " + HELM_VERSION);
```
[the remark this answers is not in the document]
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/kubernetes/operator/BaseTest.java]: Modified to   System.out.println("Helm Version is " + HELM_VERSION)

File: integration-tests/src/test/java/oracle/kubernetes/operator/ItUsabilityOperatorHelmChart.java [hunk 4]
```diff
@@ -237,13 +244,20 @@ public void testNotPreexistedOpServiceAccountCreateOperatorNegativeInstall() thr
...
           "FAILURE: Helm installs operator with not preexisted service account ");
 
     } catch (Exception ex) {
-      String cmdLb = "helm list --failed " + "  | grep " + oprelease;
+      String cmdLb = "";
+      if (BaseTest.HELM_VERSION.equals("V2")) {
+        cmdLb = "helm list --failed " + "  | grep " + oprelease;
+      }
+      if (BaseTest.HELM_VERSION.equals("V3")) {
+        cmdLb = "helm list --failed --namesapce " + opnamespace + "  | grep " + oprelease;
```
[the remark this answers is not in the document]
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/kubernetes/operator/ItUsabilityOperatorHelmChart.java]: done

File: integration-tests/src/test/java/oracle/kubernetes/operator/ItUsabilityOperatorHelmChart.java [hunk 5]
```diff
@@ -430,7 +459,13 @@ public void testSecondOpSharingSameExternalRestPortNegativeInstall() throws Exce
             "FAILURE: Helm install operator with dublicated rest port number does not report expected message "
                 + ex.getMessage());
       }
-      String cmdLb = "helm list --failed " + "  | grep " + oprelease;
+      String cmdLb = "";
+      if (BaseTest.HELM_VERSION.equals("V2")) {
+        cmdLb = "helm list --failed " + "  | grep " + oprelease;
+      }
+      if (BaseTest.HELM_VERSION.equals("V3")) {
+        cmdLb = "helm list --failed --namesapce " + opnamespace + "  | grep " + oprelease;
```
[the remark this answers is not in the document]
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/kubernetes/operator/ItUsabilityOperatorHelmChart.java]: done

File: integration-tests/src/test/java/oracle/kubernetes/operator/utils/LoadBalancer.java [hunk 6]
```diff
@@ -36,7 +36,12 @@ public LoadBalancer(Map lbMap) throws Exception {
         Paths.get(userProjectsDir + "/load-balancers/" + lbMap.get("domainUID")));
 
     if (lbMap.get("loadBalancer").equals("TRAEFIK")) {
-      String cmdLb = "helm list traefik-operator | grep DEPLOYED";
+      String cmdLb = "Empty String";
```
[the remark this answers is not in the document]
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/kubernetes/operator/utils/LoadBalancer.java]: reset the string to String cmdLb = ""

File: integration-tests/src/test/java/oracle/kubernetes/operator/utils/LoadBalancer.java [hunk 7]
```diff
@@ -53,7 +58,12 @@ public LoadBalancer(Map lbMap) throws Exception {
     }
 
     if (lbMap.get("loadBalancer").equals("VOYAGER")) {
-      String cmdLb = "helm list voyager-operator | grep DEPLOYED";
+      String cmdLb = "Empty String";
+      if (! BaseTest.HELM_VERSION.equals("V2")) {
```
[the remark this answers is not in the document]
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/kubernetes/operator/utils/LoadBalancer.java]: Changed to   if ( BaseTest.HELM_VERSION.equals("V2")) {

File: integration-tests/src/test/java/oracle/kubernetes/operator/utils/LoadBalancer.java [hunk 8]
```diff
@@ -163,7 +171,11 @@ private void upgradeTraefikNamespace() throws Exception {
    * @throws Exception when could not get values
    */
   private String getKubernetesNamespaceToUpdate(String domainNamespace) throws Exception {
-    ExecResult result = TestUtils.exec("helm get values traefik-operator", true);
+    String cmd = new String("helm get values traefik-operator ");
+    if (! BaseTest.HELM_VERSION.equals("V2")) {
+      cmd = cmd + new String(" --namespace traefik ");
```
[the remark this answers is not in the document]
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/kubernetes/operator/utils/LoadBalancer.java]: Modified as follows  ...
String cmd = "helm get values traefik-operator ";
    if (! BaseTest.HELM_VERSION.equals("V2")) {
      cmd = cmd + " --namespace traefik ";
    }

File: integration-tests/src/test/java/oracle/kubernetes/operator/utils/LoadBalancer.java [hunk 9]
```diff
@@ -189,23 +201,31 @@ private String getKubernetesNamespaceToUpdate(String domainNamespace) throws Exc
...
-        .append(" --set ")
-        .append("wlsDomain.clusterName=")
-        .append(lbMap.get("clusterName"))
-        .append(" --set ")
-        .append("traefik.hostname=")
-        .append(lbMap.get("domainUID"))
+    cmd.append(chartDir).append(" && ");
+    
+    if (BaseTest.HELM_VERSION.equals("V2")) {
+      cmd.append(" helm install ingress-per-domain ")
```
[the remark this answers is not in the document]
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/kubernetes/operator/utils/LoadBalancer.java]: Helm V3.x  does not accept --name argument, instead it take a chart name as first argument directly.  So there is no common syntax for both Helm Release

File: integration-tests/src/test/java/oracle/kubernetes/operator/utils/LoadBalancer.java (hunk 1 above)
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/kubernetes/operator/utils/LoadBalancer.java]: done

File: integration-tests/src/test/java/oracle/kubernetes/operator/utils/LoadBalancer.java (hunk 2 above)
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/kubernetes/operator/utils/LoadBalancer.java]: There is no common syntax for both Helm Release

File: src/integration-tests/bash/cleanup.sh [hunk 10]
```diff
@@ -317,11 +317,30 @@ mkdir -p $TMP_DIR || fail No permision to create directory $TMP_DIR
...
+     helm list --short --namespace $ns | while read helm_name; do
+     if [ "$HELM_VERSION" == "V2" ]; then
+       helm delete --purge  $helm_name
+     else 
+      helm uninstall $helm_name -n $ns 
+     fi
+     done
   done
 
+  # Remove namespace(s) from usablity Integration tests
```
Comment (vanajamukkara) [inline src/integration-tests/bash/cleanup.sh]: do this in the IT class in unprepare rather than having test specific code in common cleanup script

Comment (anpanigr) [reply src/integration-tests/bash/cleanup.sh]: Removed

**Comment by markxnelson** [comment]
Please address the outstanding issues and get a clean test run before requesting merge.

**Comment by markxnelson** [comment]
@rjeberhard i am ok to merge this - i have seen a satisfactory test run.  if you are ok with this, i will go ahead and merge

**Pull request merged**